adding new settings for hostname_verification (#10543)
opensearch-project/documentation-website: _security/configuration/tls.md

Name | Description
:--- | :---
`transport.ssl.enforce_hostname_verification` | Whether to verify hostnames on the transport layer. Optional. Default is `true`.
`plugins.security.ssl.transport.enforce_hostname_verification` (Deprecated) | This setting has been deprecated. Use `transport.ssl.enforce_hostname_verification` instead.

> `transport.ssl.resolve_hostname` | Whether to resolve hostnames using DNS at the transport layer. Optional. Default is `true`. Only works if hostname verification is enabled.

Reviewer: ```suggestion
`transport.ssl.resolve_hostname` | Whether to resolve hostnames using DNS on the transport layer. Optional. Default is `true`. Only works if hostname verification is enabled.
```

> `transport.ssl.resolve_hostname` | Whether to resolve hostnames against DNS on the transport layer. Optional. Default is `true`. Only works if hostname verification is also enabled.

Reviewer 2: ```suggestion
`transport.ssl.resolve_hostname` | Whether to resolve hostnames using DNS at the transport layer. Optional. Default is `true`. Only works if hostname verification is enabled.
```
